extra_tests/snippets/stdlib_time.py

```
@@ -17,3 +17,35 @@
+        ("mktime", lambda c=case: time.mktime(c)),
+        ("asctime", lambda c=case: time.asctime(c)),
+        ("strftime", lambda c=case: time.strftime("%Y", c)),
+    ]:
+        try:
+            call()
+        except OverflowError:
+            pass  # expected, matches CPython
+        except TypeError as e:
+            raise AssertionError(
+                f"{func_name}({case}) raised TypeError (should be OverflowError): {e}"
+            )
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)